### PR TITLE
feat:object agnostic design for apex code for Customer List component

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

Modify apex code for CustomerList component to use custom metadata and not reference object names as per Object agnostic design pattern

### What issues does this PR fix or reference?

This PR improves the Customer List component to completely object agnostic from the backend.

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

**Comment by msrivastav13** [comment]
@pozil We document in repo about object agnostic design for customerList component. Looks like we only follow on front end only and not on the backend apex that references objects in code.

An object agnostic approach should not reference the objects. So I converted the backend code to use custom metadata mapper and not reference them in code.

We might do the same for flows in winter 21 updates https://releasenotes.docs.salesforce.com/en-us/winter21/release-notes/rn_forcecom_flow_extend_multiple_objects.htm to showcase devs they can loosely couple the components using custom metadata.

File: es-space-mgmt/main/default/classes/reservationManagerController.cls [hunk 1]
```diff
@@ -5,54 +5,50 @@ public with sharing class reservationManagerController {
...
+            WHERE Sobject_Type__r.QualifiedAPIName = :sObjectType
+            WITH SECURITY_ENFORCED
+            LIMIT 1
+        ];
+
+        List<String> draftStatus = new List<String>();
+        draftStatus = c.Customer_Draft_Status_Values__c.split(',');
+
+        String query = 'SELECT ';
+        query = query + c.Customer_Name__r.QualifiedAPIName + ', ';
```
Comment (pozil) [inline es-space-mgmt/main/default/classes/reservationManagerController.cls]: Can you replace all of those with the `+=` operator?

File: es-space-mgmt/main/default/classes/reservationManagerController.cls [hunk 2]
```diff
@@ -5,54 +5,50 @@ public with sharing class reservationManagerController {
...
-                    c.Reservation_Status__c,
-                    c.Id
-                );
-                customers.add(cu);
-            }
+        query = query + ' WITH SECURITY_ENFORCED LIMIT 5';
+
+        for (Sobject sobj : Database.query(query)) {
+            customerServices.Customer customer = new customerServices.Customer(
+                (String) sobj.get(c.Customer_Name__r.QualifiedAPIName),
```
Comment (pozil) [inline es-space-mgmt/main/default/classes/reservationManagerController.cls:43]: Shouldn't we catch/throw an exception on these lines if the fields do no exist or if we do not have access to them?

**Review by pozil — CHANGES_REQUESTED** [review]
Good call @msrivastav13. Can you make some small changes and let me know what you think in terms of error handling?

File: es-space-mgmt/main/default/classes/reservationManagerController.cls (hunk 2 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/classes/reservationManagerController.cls:43]: @pozil The error handling is missing everywhere in this application! I can take a stab at it for this class and may be we handle for rest as we touch base the other parts of the app.

Comment (pozil) [reply es-space-mgmt/main/default/classes/reservationManagerController.cls:43]: Ok fair enough, don't bother then.

**Review by pozil — APPROVED** [review]

**Pull request merged**